File: MyAppOne/MyAppOne/Model/MAOListViewControllerModel.m
//
//  MAOListViewControllerModel.m
//  MyAppOne
//
//  Created by Julio Castillo on 10/1/19.
//  Copyright © 2019 iOS-accelerator. All rights reserved.
//

#import "MAOListViewControllerModel.h"

@implementation MAOListViewControllerModel

- (instancetype)initFromDictionary:(NSDictionary *)dictionary {
    self = [super init];
    if(self) {
        _trackId = [dictionary valueForKey:@"trackId"];
        _artistName = [dictionary valueForKey:@"artistName"];
        _collectionName = [dictionary valueForKey:@"collectionName"];
        _artworkUrl100 = [dictionary valueForKey:@"artworkUrl100"];
        _trackName = [dictionary valueForKey:@"trackName"];
        _artistViewUrl = [dictionary valueForKey:@"artistViewUrl"];
        _collectionViewUrl = [dictionary valueForKey:@"collectionViewUrl"];
        _trackViewUrl = [dictionary valueForKey:@"trackViewUrl"];
        _collectionPrice = [dictionary valueForKey:@"collectionPrice"];
        _trackPrice = [dictionary valueForKey:@"trackPrice"];
        _previewUrl = [dictionary valueForKey:@"previewUrl"];
        
        NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
        [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:00Z"];
        NSDate *theDate = [dateFormatter dateFromString:[dictionary valueForKey:@"releaseDate"]];
        _releaseDate = theDate;
    }
    return self;
}

@end

File: MyAppOneSwift/MyAppOneSwift/Model/MAOListViewControllerModel.swift
//
//  MAOListViewControllerModel.swift
//  MyAppOneSwift
//
//  Created by Roman Guarino on 25/01/2019.
//  Copyright © 2019 Román Guarino. All rights reserved.
//

import Foundation

class MAOListViewControllerModel : Decodable {
    var trackId: Int?
    var artistName: String?
    var collectionName: String?
    var trackName: String?
    var artistViewUrl: String?
    var collectionViewUrl: String?
    var artworkUrl100: String?
    var trackViewUrl: String?
    var collectionPrice: Float?
    var trackPrice: Float?
    var releaseDate: String?
    var previewUrl: String?
    
    convenience init(dict: Dictionary<String, AnyObject>) {
        self.init()
        
        self.trackId = dict["trackId"] as? Int
        self.artistName = dict["artistName"] as? String
        self.collectionName = dict["collectionName"] as? String
        self.trackName = dict["trackName"] as? String
        self.artistViewUrl = dict["artistViewUrl"] as? String
        self.collectionViewUrl = dict["collectionViewUrl"] as? String
        self.artworkUrl100 = dict["artworkUrl100"] as? String
        self.trackViewUrl = dict["trackViewUrl"] as? String
        self.collectionPrice = dict["collectionPrice"] as? Float
        self.trackPrice = dict["trackPrice"] as? Float
        self.releaseDate = dict["releaseDate"] as? String
        self.previewUrl = dict["previewUrl"] as? String
    }
}

